examples/c/ex_col_store.c

```diff
@@ -0,0 +1,582 @@
+    error_check(
+      session->open_cursor(session, "index:weather:hour", NULL, NULL, &start_time_cursor));
+    error_check(session->open_cursor(session, "index:weather:hour", NULL, NULL, &end_time_cursor));
+
+    /* select values WHERE (hour >= start AND hour <= end) */
+    /* Find the starting record closest to desired start time. */
+    start_time_cursor->set_key(start_time_cursor, start_time);
+    error_check(start_time_cursor->search_near(start_time_cursor, &exact));
+    ret = 0;
+    if (exact == -1)
+        ret = start_time_cursor->next(start_time_cursor);
+
+    if (ret == 0)
+        error_check(session->join(session, join_cursor, start_time_cursor, "compare=ge"));
+    else
+        return ret;
+
+    /* Find the ending record closest to desired end time. */
+    end_time_cursor->set_key(end_time_cursor, end_time);
+    error_check(end_time_cursor->search_near(end_time_cursor, &exact));
+    if (exact == 1)
+        ret = end_time_cursor->prev(end_time_cursor);
+
+    if (ret == 0)
+        error_check(session->join(session, join_cursor, end_time_cursor, "compare=le"));
+    else
+        return ret;
+
+    /* Initialize maximum temperature to temperature of the first record. */
+    join_cursor->next(join_cursor);
+    if (ret != 0)
+        return ret;
+
+    error_check(join_cursor->get_key(join_cursor, &recno));
+    error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+    *max_temp = temp;
+
+    /* Iterating through found records between start and end time to find the maximum temperature.
+     */
+    while ((ret = join_cursor->next(join_cursor)) == 0) {
+        error_check(join_cursor->get_key(join_cursor, &recno));
+        error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+
+        if (temp > *max_temp)
+            *max_temp = temp;
+
+        /* For debugging purposes */
+        /*
+            printf("ID %" PRIu64, recno);
+            printf(": hour %" PRIu16 " temp: %" PRIu8 "\n", hour, temp);
+         */
+    }
+
+    return (EXIT_SUCCESS);
+}
+
+/* Obtains the average data across all fields given a specific location. */
+void
+average_data(WT_SESSION *session)
+{
+    WT_CURSOR *loc_cursor;
+    const char *day;
+    const char *country;
+    uint64_t recno;
+    uint16_t hour, pressure, loc_long, loc_lat;
+    uint8_t temp, humidity, wind, feels_like_temp;
+    unsigned int count;
+    /* num_rec is the total number of records we're obtaining averages for. */
+    int ret, num_rec;
+    /* rec_arr holds the sum of the records in order to obtain the averages. */
+    unsigned int rec_arr[5] = {0, 0, 0, 0, 0};
+
+    /* Open a cursor to search for the location, currently RUS. */
+    error_check(session->open_cursor(session, "index:weather:country", NULL, NULL, &loc_cursor));
+    loc_cursor->set_key(loc_cursor, "RUS\0\0");
+    ret = loc_cursor->search(loc_cursor);
+
+    /* Error handling in the case RUS is not found. In this case as it's a hardcoded location,
```

Comment:
```suggestion
    /*
     * Error handling in the case RUS is not found. In this case as it's a hardcoded location,
     * if there aren't any matching locations, no average data is obtained and we proceed with the
     * test instead of aborting. If an unexpected error occurs, exit.
     */
```

Reply:
Sorry, I don't really understand the change - did you want to change the wording of the comment?

Reply:
No, I just pointed out the way the multi-line comment should be. No changes to the comment content.